src/sortStudents.ts

```diff
@@ -1,16 +1,66 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+const AverageGrade
+  = (arr: number[]): number => {
+    if (arr.length === 0) {
+      return 0;
+    }
 
+    return arr.reduce((a: number, b: number) => a + b, 0) / arr.length;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+  const orderType = order === 'asc';
+
+  if (sortBy === SortType.Name) {
+    return orderType
+      ? studentsCopy.sort((a, b) => a.name.localeCompare(b.name))
+      : studentsCopy.sort((a, b) => b.name.localeCompare(a.name));
+  }
+
+  if (sortBy === SortType.Surname) {
+    return orderType
+      ? studentsCopy.sort((a, b) => a.surname.localeCompare(b.surname))
+      : studentsCopy.sort((a, b) => b.surname.localeCompare(a.surname));
+  }
```

Comment:
1. please use [switch case](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) for it 
2.  your cases can be grouped - 2 numeric, 2 string and 1 avarage 

```suggestion
case SortType.Name:
case SortType.Surname:
   return (order === 'asc')
      ? studentsCopy.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
      : studentsCopy.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
```